src/read_pair_iter.rs

```diff
@@ -352,6 +382,7 @@ impl ReadPairIter {
         if self.buffer.remaining_mut() < 512 {
             self.buffer = BytesMut::with_capacity(BUF_SIZE)
         }
+        let is_single_ended = self.is_single_ended()?;
```

Comment:
This function is inside the "inner loop" of FASTQ processing, so you can't be re-running the fairly heavy-weight `is_single_ended` function here -- it's too slow because it's opening files and parsing reads.

I thinkis_single_ended should be a pre-computed flag that is set when you construct the ReadPairIter.

Reply:
Fixed. This had made the iterations 100x slower on testing (`DETECT_CHEMISTRY` that usually takes <5 min had run overnight for something like 8hours when I was testing).